block-producer/src/test_utils/proven_tx.rs

```diff
@@ -33,29 +24,58 @@ static NUM_NOTES_CREATED: Lazy<Arc<Mutex<u64>>> = Lazy::new(|| Arc::new(Mutex::n
 static NUM_INPUT_NOTES: Lazy<Arc<Mutex<u64>>> = Lazy::new(|| Arc::new(Mutex::new(0)));
 
 pub struct MockProvenTxBuilder {
-    mock_account: MockPrivateAccount,
+    account_id: AccountId,
+    initial_account_hash: Digest,
+    final_account_hash: Digest,
     notes_created: Option<Vec<NoteEnvelope>>,
     nullifiers: Option<Vec<Nullifier>>,
 }
 
 impl MockProvenTxBuilder {
     pub fn new() -> Self {
-        let account_index: u32 = {
+        let mock_account: MockPrivateAccount = {
             let mut locked_num_accounts_created = NUM_ACCOUNTS_CREATED.lock().unwrap();
```

Comment:
I'm against this approach. This is adding synchronization among tests. The results of these values will depend on the number and order the tests are executed. Which can easily lead to brittle tests and flakiness.

IMO these values should be inject.